libbpf-rs/src/object.rs

```diff
@@ -334,6 +353,58 @@ impl Object {
         self.map(name).unwrap().unwrap()
     }
 
+
+    pub fn get_maps(&mut self) -> Result<&HashMap<String, Map>> {
+
+        // if populated already return the populated hashmap
+        if !self.maps.is_empty() {
+            Ok(&self.maps)
+
+        } else {
+     
+            let mut maps: HashMap<String, Map> = HashMap::new();
+            let mut map: *mut libbpf_sys::bpf_map = std::ptr::null_mut();
+
+            loop {
+                // while there's another map to grab
+                if let Some(next_map_ptr) = object_map_next(self.ptr, map).unwrap() {
```

Comment:
See above, we don't need to return a `Ok(..)` and `unwrap()`. No reason to do extra work